swarm/network/kademlia.go

```diff
@@ -741,34 +760,41 @@ func (k *Kademlia) gotNearestNeighbours(peers [][]byte) bool {
 		pm[pk] = true
 		return true
 	})
+	var gots int
 	for _, p := range peers {
 		pk := fmt.Sprintf("%x", p)
-		if !pm[pk] {
+		if pm[pk] {
+			gots++
+		} else {
 			log.Trace(fmt.Sprintf("%08x: ExpNN: %s not found", k.BaseAddr()[:4], pk[:8]))
-			return false
 		}
 	}
-	return true
+	if gots < len(pm) {
```

Comment:
I believe we are generally missing comments about such important logic decisions; can we have one here, why is this condition needed? e.g "only true if minProxBinSize reached" (or whatever is the reason)

Reply:
Oh wow, there probably shouldn't be changes in there. Thanks.

Reply:
Turns out there was a blatant logic fault there. It needs to be `len(peers)` to be equivalent.